bundles/org.openhab.binding.samsungtv/README.md

```diff
@@ -26,81 +31,639 @@ Tested TV models:
+
+If you enable the Smartthings interface, this adds back the `sourceName`, `sourceId`, `programTitle` and `channelName` channels on >2016 TV's
+Samsung removed the app API support in >2019 TV's, if your TV is >2019, see the section on [Apps](#apps).
+Samsung removed the art API support in >2021 TV's, if your TV is >2021, see the section on [setArtMode](#setArtMode) .
+
+**NOTE:** `brightness`, `contrast`, `colorTemperature` and `sharpness` channels only work on legacy interface TV's (<2016).
 
 ## Discovery
 
-The TV's are discovered through UPnP protocol in the local network and all devices are put in the Inbox.
+The TV's are discovered through UPnP protocol in the local network and all devices are put in the Inbox. TV must be ON for this to work.
 
 ## Binding Configuration
 
-The binding does not require any special configuration.
+Basic operation does not require any special configuration.  
+For >2019 TV's, the app workaround uses a built in list of known app ID's. You can add to this list by editing the `misc/samsungtv.applist` file in the openhab config directory.  
+To enable Smartthings polling, you have to add a Smartthings Personal Access token (PAT) to the Thing configuration.  
 
 ## Thing Configuration
 
-The Samsung TV Thing requires the host name and port address as a configuration value in order for the binding to know how to access it.
-Samsung TV publish several UPnP devices and hostname is used to recognize those UPnP devices.
-Port address is used for remote control emulation protocol.
-Additionally, a refresh interval can be configured in milliseconds to specify how often TV resources are polled.
+The Samsung TV Thing requires the host name and port address as a configuration value in order for the binding to know how to access it.  
+Samsung TV's publish several UPnP devices and the hostname is used to recognize those UPnP devices.  
+Port address is used for remote control emulation protocol.  
+Additionally, a refresh interval can be configured in milliseconds to specify how often TV resources are polled. Default is 1000 ms.  
 
 E.g.
 
-```java
+```
 Thing samsungtv:tv:livingroom [ hostName="192.168.1.10", port=55000, macAddress="78:bd:bc:9f:12:34", refreshInterval=1000 ]
 ```
 
-Different ports are used in different models. It may be 55000, 8001 or 8002.
+Different ports are used on different models. It may be 55000, 8001 or 8002.
+
+If you have a <2016 TV, the interface will be *Legacy*, and the port is likely 55000.  
+If you have a >2016 TV, the interface will be either *websocket* on port 8001, or *websocketsecure* on port 8002.  
+If your TV surrports *websocketsecure*, you **MUST** use it, otherwise the `keyCode` channel will not work, and as a lot of internal logic relies on that channel, you will find many things do not work (`power` for example).  
```

Comment:
```suggestion
If your TV supports *websocketsecure*, you **MUST** use it, otherwise the `keyCode` channel will not work, and as a lot of internal logic relies on that channel, you will find many things that do not work (`power` for example).  
```

Reply:
Fixed